lib/spack/spack/test/conftest.py

```diff
@@ -63,6 +64,59 @@ def no_stdin_duplication(monkeypatch):
                         lambda x: StringIO())
 
 
+@pytest.fixture()
+def spec_from_dict():
```

Comment:
There's no need to make this a fixture.  Why require it?  It should just be a static method on `Spec`, and it's generally useful.  How about `Spec.from_dict()`?